### PR TITLE
[MIG] date_range: Migrate to version 17.0

### PR DESCRIPTION
**Pull request by sonhd91**

### CODE REVIEW

**Comment by dzungtran89** [comment]
I got this error as running the test
```
psycopg2.errors.NotNullViolation: null value in column "fiscalyear_last_day" of relation "res_company" violates not-null constraint
```

**Comment by sonhd91** [comment]
> I got this error as running the test
> 
> ```
> psycopg2.errors.NotNullViolation: null value in column "fiscalyear_last_day" of relation "res_company" violates not-null constraint
> ```
I tested it on my local environment, and all tests passed. The `fiscalyear_last_day` field is defined in the `account` module, which means this module doesn't need to depend on it. Could you please test this on a fresh db?

**Comment by dzungtran89** [comment]
An exception raises if the type does not have any date ranges, could you please help?

**Comment by dzungtran89** [comment]
I tried to inactive the range type, but its ranges are still visible, could you help to check?

File: date_range/models/date_range_type.py [hunk 1]
```diff
@@ -83,7 +83,8 @@ def _check_company_id(self):
                     continue
                 if bool(
                     rec.date_range_ids.filtered(
-                        lambda r: r.company_id and r.company_id != rec.company_id
+                        lambda r, drt=rec: r.company_id
```
Comment (dzungtran89) [inline date_range/models/date_range_type.py:86]: This violates B023, `any([range_date.company_id and range_date.company_id != rec.company_id for range_date in rec.date_range_ids])` can make the ruff lint happy

Comment (dzungtran89) [reply date_range/models/date_range_type.py:86]: hmm, somehow the pre-commit could pass :thinking:

**Review by dzungtran89 — APPROVED** [review]

**Pull request merged**